web/app/scripts/filterbar/filterbar-controller.js

```diff
@@ -54,9 +61,11 @@
         /**
          * When the record type changes, request the new schema
          */
-        $scope.$on('driver.state.recordstate:selected', function(event, selected) {
-            if (selected) {
+        $scope.$on('driver.state.recordstate:selected',
+                   function(event, selected) { onRecordSelected(selected); });
```

Comment:
I think you might be misreading it:
function(event, selected) { onRecordSelected(selected); });
This will take an `event` (first arg) and `selected` (second arg), apply onRecordSelected with only the second arg and delegate `event` to garbage collection.